### PR TITLE
Misc addons 10.0 ir attachment url

### PR DESCRIPTION
**Pull request by iledarn**

### CODE REVIEW

File: ir_attachment_url/models/ir_http.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+                        last_update = str(os.path.getmtime(module_resource_path))
+
+            if not module_resource_path:
+                module_resource_path = obj.url
+
+            if not content:
+                status = 301
+                content = module_resource_path
+        else:
+            content = obj[field] or ''
```
Comment (yelizariev) [inline ir_attachment_url/models/ir_http.py]: don't it call whenever it's not required

**Comment by iledarn** [comment]
@yelizariev, done

File: ir_attachment_url/models/ir_http.py [hunk 2]
```diff
@@ -89,6 +90,8 @@ def binary_content(cls, xmlid=None, model='ir.attachment', id=None, field='datas
                 if att:
                     content = att.url
                     status = 301
+            if not is_attachment or not att:
```
Comment (yelizariev) [inline ir_attachment_url/models/ir_http.py]: if not content:

**Comment by iledarn** [comment]
@yelizariev , done

File: ir_attachment_url/models/ir_http.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        etag = hasattr(request, 'httprequest') and request.httprequest.headers.get('If-None-Match')
+        retag = '"%s"' % hashlib.md5(last_update).hexdigest()
+        status = status or (304 if etag == retag else 200)
+        headers.append(('ETag', retag))
+        headers.append(('Cache-Control', 'max-age=%s' % (STATIC_CACHE if unique else 0)))
+
+        # content-disposition default name
+        if download:
+            headers.append(('Content-Disposition', cls.content_disposition(filename)))
+        print '\n\n\n\n\n', 'headers HEADERS ', headers, '\n\n\n\n'
```
Comment (yelizariev) [inline ir_attachment_url/models/ir_http.py]: delete prints

**Comment by yelizariev** [comment]
@iledarn fix lints

ir_attachment_s3/models/ir_attachment.py:3:1: F401 'urlparse' imported but unused
ir_attachment_s3/models/ir_attachment.py:13:1: E402 module level import not at top of file
ir_attachment_s3/models/ir_attachment.py:13:1: F401 'odoo.fields' imported but unused
ir_attachment_s3/models/ir_attachment.py:14:1: E402 module level import not at top of file
ir_attachment_s3/models/ir_attachment.py:14:1: F401 'odoo.exceptions.UserError' imported but unused
ir_attachment_s3/models/ir_attachment.py:15:1: E402 module level import not at top of file
ir_attachment_s3/models/ir_attachment.py:51:17: E126 continuation line over-indented for hanging indent
ir_attachment_s3/models/ir_attachment.py:76:21: E126 continuation line over-indented for hanging indent
ir_attachment_url/models/ir_attachment.py:4:1: F401 'odoo.fields' imported but unused
ir_attachment_url/models/ir_http.py:6:1: F401 'openerp.fields' imported but unused
ir_attachment_url/models/ir_http.py:9:1: F401 'odoo.http.content_disposition' imported but unused
ir_attachment_url/models/ir_http.py:51:16: F821 undefined name 'AccessError'
ir_attachment_url/models/ir_http.py:59:25: F821 undefined name 're'
ir_attachment_url/models/ir_http.py:62:31: F821 undefined name 'get_module_path'
ir_attachment_url/models/ir_http.py:63:40: F821 undefined name 'get_resource_path'
ir_attachment_url/models/ir_http.py:65:35: F821 undefined name 'os'
ir_attachment_url/models/ir_http.py:65:48: F821 undefined name 'os'
ir_attachment_url/models/ir_http.py:66:44: F821 undefined name 'os'
ir_attachment_url/models/ir_http.py:70:43: F821 undefined name 'os'
ir_attachment_url/models/ir_http.py:101:28: F821 undefined name 'os'


commit message I will use
[ADD]  ir_attachment_s3, ir_attachment_url (it-projects-llc/addons-dev#161)

**Comment by yelizariev** [comment]
add boto3 to requirements.rst

**Comment by yelizariev** [comment]
fix

ir_attachment_url/models/ir_http.py:59: [W1401(anomalous-backslash-in-string), ] Anomalous backslash in string: '\w'. String constant might be missing an r prefix.

**Comment by yelizariev** [comment]
fix

ir_attachment_url/README.rst:1: [W7908(missing-newline-extrafiles), ]  Missing newline

**Pull request merged**